sound/soc/sof/control.c

```diff
@@ -11,8 +11,38 @@
 /* Mixer Controls */
 
 #include <linux/pm_runtime.h>
+#include <linux/leds.h>
 #include "sof-priv.h"
 
+static void update_mute_led(struct snd_sof_control *scontrol,
+			    struct snd_kcontrol *kcontrol,
+			    struct snd_ctl_elem_value *ucontrol)
+{
+	unsigned int temp = 0;
+	unsigned int mask;
+	int i;
+
+	mask = 1U << snd_ctl_get_ioffidx(kcontrol, &ucontrol->id);
+
+	temp &= ~mask;
+	for (i = 0; i < scontrol->num_channels; i++) {
+		if (ucontrol->value.integer.value[i]) {
+			temp |= mask;
```

Comment:
is there a 'i' missing somewhere, or are we going to 'or' the same thing multiple times? If this is intentional please comment.

Reply:
@plbossart we will break out of the loop the first we hit a non-zero value[i]

Reply:
ah yes, discard this comment then